cetloxblock/cetloxblock.py

```diff
@@ -0,0 +1,138 @@
+        scope=Scope.settings,
+        default=u"CET LO Activity")
+
+    documentid = String(default="", scope=Scope.content, help="The LO id")
+    language = String(default="he", scope=Scope.content, help="The LO language code")
+    folderid = String(default="", scope=Scope.content, help="The LO folder (scope) id for adaptive LOs")
+    taskid = String(default="", scope=Scope.user_state, help="The LO id")
+
+    has_author_view = True
+    
+    def resource_string(self, path):
+        """ helper for getting UNICODE STRINGS from static resources (from readthedocs/xblock tutorial)."""
+        data = pkg_resources.resource_string(__name__, path)
+        data = data.decode("utf8")
+        return data
+
+    def author_view(self, context=None):
+        if not self.documentid:
+            return Fragment(u"<h1>Please edit to select an activity for this component</h1>")
+
+        return self.student_view(context)
+
+    def student_view(self, context=None):
+        """
+        The primary view of the CetLoXBlock, shown to students when viewing courses.
+        """
+        if not self.documentid:
+            return Fragment(u"EMPTY student_view")
+
+        iframe_src = u"about:blank"
+        
+        itemTemplate = self.ITEM_URL_TEMPLATE.format(protocol=self.PROTOCOL, cetdomain=u"{cetdomain}", language=self.language, documentid=u"{itemid}")
+        taskTemplate = self.TASK_URL_TEMPLATE.format(protocol=self.PROTOCOL, cetdomain=u"{cetdomain}", taskid=u"{taskid}")
+        html = self.resource_string("static/html/cetlo_student_view.html")
+        html = html.format(documentid=self.documentid, language=self.language, folderid=self.folderid, taskid=self.taskid, itemTemplate=itemTemplate, taskTemplate=taskTemplate, iframeSrc=iframe_src)
+
+        frag = Fragment(html)
+        frag.add_css(self.resource_string("static/css/cetloxblock_student.css"))
+        frag.add_javascript(self.resource_string("static/js/env-manager.js"))
+        frag.add_javascript(self.resource_string("static/js/sso.js"))
+        frag.add_javascript(self.resource_string("static/js/student-xblock.js"))
+        frag.add_javascript(self.resource_string("static/js/services.js"))
+        frag.initialize_js('CetLoXBlock')
+        return frag
+
+    def studio_view(self, context=None):
+        """
+        The studio view of the CetLoXBlock, letting author to select a LO.
+        This is a stub showing a fix list of LO items.
+        """
+        html = self.resource_string("static/html/cetlo_studio_view.html")
+
+        selectedLO = dict(id=self.documentid, language=self.language, folderid=self.folderid, title=self.display_name)
+        html = html.format(selected=selectedLO)
+        html += self.resource_string("static/html/cetlo_studio_item_template.html")
+
+        frag = Fragment(html)
+        frag.add_css(self.resource_string("static/css/cetloxblock_studio.css"))
+        frag.add_javascript(self.resource_string("static/js/studio-xblock.js"))

```

Comment:
Missing these JS files when editing in Studio:
```python
frag.add_javascript(self.resource_string("static/js/env-manager.js"))
frag.add_javascript(self.resource_string("static/js/sso.js"))
```

Reply:
Right. 
The 1st one added. I don't see need for sso.js in studio view.